### PR TITLE
Remove gpupgrade machinery

### PR DESCRIPTION
**Pull request by hlinnaka**
We're not going to use this mechanism for upgrading from 4.3 to 5.0, so remove it to make merging easier. Anything more that could be removed now that we don't need to support this?

### CODE REVIEW

File: src/backend/access/transam/xlog.c [hunk 1]
```diff
@@ -3367,45 +3367,6 @@ RecordIsValid(XLogRecord *record, XLogRecPtr recptr, int emode)
...
-		}
-
-		/* Finally include the record header */
-		COMP_CRC32(crc, (char *) record + sizeof(pg_crc32),
-				   SizeOfXLogRecord - sizeof(pg_crc32));
-		FIN_CRC32(crc);
-	}
-
-	if (!EQ_CRC32(record->xl_crc, crc))
-	{
```
Comment (asimrp) [inline src/backend/access/transam/xlog.c]: It's not clear to my how this removed chunk was related to upgrade.

Comment (ashwinstar) [inline src/backend/access/transam/xlog.c]: There is no way now one would find xlog record with older crc algorithm and hence this piece of code becomes obsolete.

File: src/backend/cdb/cdbpersistentbuild.c [hunk 2]
```diff
@@ -1003,123 +1003,6 @@ gp_persistent_reset_all(PG_FUNCTION_ARGS)
...
-												&gpRelationNode->persistentTid,
-												gpRelationNode->persistentSerialNum,
-												relBufpoolKind);
-		}
-
-	}
-}
-
-Datum
-gp_persistent_set_relation_bufpool_kind_all(PG_FUNCTION_ARGS)
```
Comment (ashwinstar) [inline src/backend/cdb/cdbpersistentbuild.c]: With this removal most likely should be able to remove following functions as well 
PersistentFileSysRelBufpoolKind_Name
PersistentFileSysObj_UpdateRelationBufpoolKind

**Comment by ashwinstar** [comment]
Also can be removed the starting points of the upgrade and maybe stuff only these might be using
gpMgmt/bin/gpmigrator_mirror
gpMgmt/bin/gpmigrator

**Comment by ashwinstar** [comment]
Awesome Thank You, complex functionality piece knocked out of development.

**Comment by hlinnaka** [comment]
Pushed, thanks for the review!

**Pull request closed without merging**